modules/bale_actor/inc/selector.h

```diff
@@ -354,10 +373,27 @@ class Selector {
   protected:
 
   public:
+    void trace_nodeID() {
+        nodeID = (int*)shmem_malloc(sizeof(int));
+
+        char *nid = getenv("SLURM_NODEID");
```

Comment:
Please also add `NULL` check. If either `nodeID` or `nid` is `NULL`, we could generate some error message and just abort.